File: sand-iam/examples/webman-business-app/sdk/consume.ts
import { SandIamClient } from '../../../sdk/typescript/src/index.js'
import { sandIam } from '../generated/sand_iam.js'

const credential = (globalThis as typeof globalThis & { process?: { env?: Record<string, string | undefined> } }).process?.env?.SAND_IAM_ACCESS_TOKEN ?? ''
if (credential === '') throw new Error('SAND_IAM_ACCESS_TOKEN 必须由服务端运行环境注入')

async function main(): Promise<void> {
  const client = new SandIamClient({
    baseUrl: (globalThis as typeof globalThis & { process?: { env?: Record<string, string | undefined> } }).process?.env?.SAND_IAM_BASE_URL ?? '',
    organizationCode: sandIam.organizationCode,
    applicationCode: sandIam.applicationCode,
    accessToken: () => credential,
  })
  await client.authorize({ apiCode: sandIam.actions.WORK_ITEM_READ, requestId: 'work-item-ts-read-001' })
}

void main()

File: sand-iam/examples/webman-business-app/sdk/consume.dart
import 'dart:io';

import '../../../sdk/dart/lib/sand_iam.dart';
import '../generated/sand_iam.dart';

Future<void> main() async {
  final token = Platform.environment['SAND_IAM_ACCESS_TOKEN'];
  if (token == null || token.isEmpty) {
    throw StateError('SAND_IAM_ACCESS_TOKEN 必须由运行环境或密钥管理系统注入');
  }
  final client = SandIamClient(
    baseUrl: Platform.environment['SAND_IAM_BASE_URL'] ?? '',
    organizationCode: sandIamOrganizationCode,
    applicationCode: sandIamApplicationCode,
    accessToken: () => token,
  );
  await client.authorize(
    apiCode: sandIamActions['WORK_ITEM_READ']!,
    requestId: 'work-item-dart-read-001',
  );
  stdout.writeln(sandIamAudience);
}
